boot/setup.sh

```diff
@@ -59,10 +68,12 @@ function setup_cloudapi {
 
     #haproxy
     for port in "${ports[@]}"; do
-        hainstances="$hainstances        server cloudapi-$port 127.0.0.1:$port check inter 10s slowstart 10s error-limit 3 on-error mark-down\n"
+        hainstances="$hainstances        server cloudapi-$port *:$port check inter 10s slowstart 10s error-limit 3 on-error mark-down\n"
     done
 
-    sed -e "s#@@CLOUDAPI_INSTANCES@@#$hainstances#g" \
+    ADMIN_IP=$(/usr/sbin/mdata-get sdc:nics | /usr/bin/json -a -c 'this.nic_tag === "admin"' | /usr/bin/json ip)
```

Comment:
This can be simplified, removing one call to json.

    ADMIN_IP=$(/usr/sbin/mdata-get sdc:nics | /usr/bin/json -a -c 'this.nic_tag === "admin"' ip)

Reply:
True. Just using the same thing we have in SAPI.

Reply:
I think even still, I'd rather do the simpler one and eventually get that into sapi as well.